systori/static/ts/src/project_list.ts

```diff
@@ -149,97 +60,66 @@ class SystoriSortButton extends HTMLElement {
             ? (this.dataset["reversed"] = "true")
             : (this.dataset["reversed"] = "false");
     }
+    toggleReversed(): void {
+        this.reversed = !this.reversed;
+    }
 
     get active(): boolean {
         return this.classList.contains("active");
     }
     set active(active: boolean) {
         active ? this.classList.add("active") : this.classList.remove("active");
     }
+    sortProjectTiles(this: SystoriSortButton): void {
+        this.toggleReversed();
 
-    activateExclusive(): void {
-        const btns = Array.from(
-            (this.parentElement as HTMLDivElement).querySelectorAll<
-                SystoriSortButton
-            >("sys-sort-button"),
+        const projectTiles = Array.from(
+            document.querySelectorAll<SystoriProjectTile>(".tile"),
         );
-        for (const btn of btns) {
-            btn.active = false;
-        }
-        this.active = true;
-        localStorage["sys-sort-button"] = this.type;
-        localStorage["sys-sort-button-reversed"] = this.reversed.toString();
-    }
-
-    connectedCallback(): void {
-        if (this.dataset["type"] != null)
-            this.type = this.dataset["type"] as string;
-        if (this.dataset["reversed"] != null)
-            this.reversed = this.dataset["reversed"] == "true";
-    }
-}
 
-class SystoriProjectTile extends HTMLElement {
-    tag = "sys-project-tile";
-
-    get hideProjectTile(): boolean {
-        return this.classList.contains("hidden");
-    }
-    set hideProjectTile(hideProjectTile: boolean) {
-        hideProjectTile
-            ? this.classList.add("hidden")
-            : this.classList.remove("hidden");
-    }
-}
-
-class SystoriWarningMessage extends HTMLElement {
-    tag = "sys-warning-message";
-
-    warnPhaseFilteredProjects(phaseFilteredProjects: number): void {
-        if (phaseFilteredProjects > 0) {
-            this.children[0].innerHTML = (document.querySelector(
-                "#sys-phaseFilteredProjects-translated",
-            ) as HTMLElement).innerText;
-            this.classList.remove("hidden");
+        projectTiles.sort((a, b) => {
+            if (this.type === "id") {
```

Comment:
Suggestion: maybe use enum for all of it's types
```suggestion
            if (this.type === SortButtonType.id) {
```

Reply:
Also, maybe use `switch` statement. The I don't personally like `if-elseif-else`. I nsiede those if there are more ifs and then finally there is `?:` operator which is also if and else
 
Switch example
```ts
        const projectTiles = Array.from(
            document.querySelectorAll<SystoriProjectTile>(".tile"),
        );

        projectTiles.sort((a, b) => {
            switch (this.type) {
                case SortButtonType.id:
					// The methods can check for `this.reversed`
                    return this.sortById(a.pk, b.pk);
                case SortButtonType.name:
                    this.sortByName(a.name, b.name);
                case SortButtonType.phase:
                    this.sortByPhase(a.phase, b.phase);
                default:
                    throw new Error("Unkown Button type.");
            }
        });

        if (tileContainer) {
            tileContainer.innerHTML = "";
            for (const tile of projectTiles) {
                tileContainer.appendChild(tile);
            }
        }

		// Once the reverse actually happens
        this.toggleReversed();
```